src/encoded/audit/experiment.py

```diff
@@ -61,37 +63,38 @@ def audit_hic_restriction_enzyme_in_libaries(value, system, excluded_types):
         return
     if 'replicates' not in value:
         return
-    frag_methods = set()
-    using_restriction_enzyme = 0
+    
+    
+    fragmentation_methods = set()
+    libraries_fragmentation_methods = {}
+    print ("zopa")
     for rep in value['replicates']:
         rep_lib = rep.get('library')
         rep_status = rep.get('status')
         if rep_lib and rep_status and rep_status not in excluded_types:
             rep_lib_status = rep_lib.get('status')
             if rep_lib_status and rep_lib_status not in excluded_types:
                 if 'fragmentation_method' in rep_lib:
-                    frag_methods.add(rep_lib['fragmentation_method'])
-                    if 'restriction' in rep_lib['fragmentation_method']:
-                        using_restriction_enzyme = 1
+                    libraries_fragmentation_methods[rep_lib.get('@id')] = set(rep_lib.get('fragmentation_method'))
+                    for method in rep_lib.get('fragmentation_method'):
+                        fragmentation_methods.add(method)
                 else:
                     detail = ('Experiment {} contains a library {} '
                               'lacking the specification of the fragmentation '
                               'method used to generate it'.format(
                                     value['@id'],
-                                    rep_lib.get('accession')
-                               )
-                              )
+                                    rep_lib.get('accession')))
                     yield AuditFailure('missing fragmentation method', detail, level='WARNING')
-
-    if len(frag_methods) > 1 and using_restriction_enzyme > 0:
-        detail = ('Experiment {} contains libraries generated '
-                  'following fragmentation with '
-                  'inconsistent restriction enzymes {} '.format(
+    for library_id in libraries_fragmentation_methods.keys():
+        methods = libraries_fragmentation_methods[library_id]
```

Comment:
You can use `libraries_fragmentation_methods.items()` instead to extract the key and value from the dictionary.

Reply:
done